h/views/activity.py

```diff
@@ -162,6 +162,27 @@ def user_annotation_count(aggregation, userid):
         return result
 
     @view_config(request_method='POST',
+                 request_param='group_join')
+    def join(self):
+        """
+        Join the given group.
+
+        This adds the authenticated user to the given group and redirect the
+        browser to the search page.
+        """
+        if not self.request.has_permission('join', self.group):
```

Comment:
Same comment here about the fact that the `self.group` param is probably not needed as `self.group` is already the request context.